lib/packer/provisioners/chef/solo.rb

```diff
@@ -9,7 +9,7 @@ class Solo < Chef
         def initialize
           super
           self.data['type'] = CHEF_SOLO
-          self.add_required([])
+          #self.add_required([])
```

Comment:
I think this is the only change we really need here, right? Don't push an empty array if there aren't any valid keys? The rest is probably unnecessary. I'm not a big fan of strong type checking ala `valid_key?` above -- I'd rather enforce as little as possible. It generally makes things more adaptable to uses you weren't anticipating.